src/sentry/taskworker/scheduler/schedules.py

```diff
@@ -127,20 +127,22 @@ def remaining_seconds(self, last_run: datetime | None = None) -> int:
         # If last run is in the past, see if the next runtime
         # is in the future.
         if last_run < now:
-            next_run = self._advance(last_run)
+            next_run = self._advance(last_run + timedelta(minutes=1))
```

Comment:
I'm a little confused by this: why always 1 minute?

Reply:
cronsim.advance() won't actually 'advance' the schedule if the current time aligns with the schedule interval. For example

```python
>>> from datetime import datetime, timedelta
>>> from cronsim import CronSim
>>> now = datetime(2024, 2, 4, 16, 50, 30)
>>> s = CronSim("*/2 * * * *", now)
>>> s.advance()
>>> s.dt
datetime.datetime(2024, 2, 4, 16, 50)
```

Because we want the *next* interval, and the smallest unit of time that Cronsim can operate on is 1 minute, we need to advance the time to get the next interval

```python
>>> now = datetime(2024, 2, 4, 16, 50, 30)
>>> s = CronSim("*/2 * * * *", now)
>>> s.advance()
>>> s.dt
datetime.datetime(2024, 2, 4, 16, 50)
```